src/lib/mongoBackend/MongoCommonRegister.cpp

```diff
@@ -269,6 +274,7 @@ static bool addTriggeredSubscriptions
       subs.insert(std::pair<string, TriggeredSubscription*>(subIdStr, trigs));
     }
   }
+  releaseMongoConnection(connection);
```

Comment:
I think that no cursor is reused in any place (i.e. the variable holding cursorr gets out of scope before that the cursor has the opportunity to be used again). However, I agree in do has much as possible to ensure this, even if that implies redundancy. Thus, I'd add a `cursor.kill()` statements just before releaseMongoConnection() calls.